src/main/java/seedu/duke/command/flashcardcommand/AddFlashcardCommand.java

```diff
@@ -1,4 +1,36 @@
 package seedu.duke.command.flashcardcommand;
 
-public class AddFlashcardCommand {
+import seedu.duke.card.*;
+import seedu.duke.exception.NoSubjectException;
+import seedu.duke.exception.RepeatedSubjectException;
+
+public class AddFlashcardCommand extends FlashcardCommand{
+	private String fullCommand;
+
+	public AddFlashcardCommand(String fullCommand) {
+		this.fullCommand = fullCommand;
+	}
+	public Flashcard execute(Topic topic) throws NoSubjectException, RepeatedSubjectException {
+		int startOfMessage = 4;
+		int endOfQuestion = (fullCommand.indexOf(";"));
+		int startOfAnswer = endOfQuestion + 2;
+		int endOfMessage = fullCommand.length();
+		if (endOfMessage <= startOfMessage) {
+			throw new NoSubjectException();
```

Comment:
Should have created new exception classes for flashcards. NoSubjectException is supposed to be used by subject commands, but I noticed that the topic commands were also using the same exceptions, so I guess we can change it later